ledger/participant-state/kvutils/tools/integrity-check/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/LogAppendingReadServiceFactory.scala

```diff
@@ -7,29 +7,30 @@ import akka.NotUsed
 import akka.stream.Materializer
 import akka.stream.scaladsl.Source
 import com.daml.ledger.api.health.HealthStatus
+import com.daml.ledger.participant.state.kvutils.OffsetBuilder
 import com.daml.ledger.participant.state.kvutils.api.{
   KeyValueParticipantStateReader,
   LedgerReader,
   LedgerRecord,
 }
 import com.daml.ledger.participant.state.kvutils.export.WriteSet
-import com.daml.ledger.participant.state.kvutils.{OffsetBuilder, Raw}
 import com.daml.ledger.participant.state.v1.{LedgerId, LedgerInitialConditions, Offset, Update}
 import com.daml.metrics.Metrics
 
 import scala.collection.immutable
 import scala.collection.mutable.ListBuffer
 
 final class LogAppendingReadServiceFactory(
-    metrics: Metrics
+    logEntryIdComputationStrategy: LogEntryIdComputationStrategy,
+    metrics: Metrics,
 ) extends ReplayingReadServiceFactory {
   private val recordedBlocks = ListBuffer.empty[LedgerRecord]
 
   override def appendBlock(writeSet: WriteSet): Unit =
     this.synchronized {
       writeSet.foreach { case (key, value) =>
         val offset = OffsetBuilder.fromLong(recordedBlocks.length.toLong)
-        val logEntryId = Raw.LogEntryId(key.bytes) // `key` is of an unknown type.
+        val logEntryId = logEntryIdComputationStrategy.compute(key, value)
```

Comment:
This will only work for ledgers not supporting privacy, i.e., for privacy-enabled ledgers we use `LogFragmentingReadServiceFactory`.